android/tests_backend/window.py

```diff
@@ -20,12 +22,24 @@ async def wait_for_window(
         message,
         minimize=False,
         full_screen=False,
-        state_switch_not_from_normal=False,
+        expected_state=None,
     ):
-        await self.redraw(
-            message,
-            delay=(0.5 if (full_screen or state_switch_not_from_normal) else 0.1),
-        )
+        await self.redraw(message, delay=(0.5 if full_screen else 0.1))
```

Comment:
What's the difference between `wait_for_window(full_screen=True)` and `wait_for_window(expected_state=FULLSCREEN)`? Looking at the tests, you've updated most (if not all) of the existing calls to use the new parameter... so why keep the old one? If it's purely because of the "cleanup" mechanism - that's something that can be replaced with a literal `self.redraw()` - essentially reproducing the old implementation of `wait_for_window` for the one place we need a "fixed" interval. Or, add a `wait_for_window_close()` probe method, if that capability is needed in more than one place.

Reply:
Yes, the old parameters are only used for `window_cleanup`. Adding a `wait_for_window_close()` probe method would allow us to remove the old parameters entirely.